designs/modify-resources-attached-to-code-pipeline.md

```diff
@@ -0,0 +1,163 @@
+* `Yes` - proceeds with the manual operation
+* `No` - The default - Cancels the current operation (or goes back a step in the case of a wizard)
+* `Learn More` - This will link to user documentation (https://docs.aws.amazon.com/codepipeline/latest/userguide) but will not close the confirmation prompt
+
+Implementation
+==============
+
+Design
+------
+
+### Resources associated with a pipeline
+
+Resources that have a tag `aws:codepipeline:pipelineArn` are considered to have an association with a pipeline.
+
+### Utility Methods
+* a method that determines if an AWS resource is associated with AWS CodePipeline
+* a method that generates confirmation prompt text (if confirmation prompt will exist within a Wizard and as a standalone prompt)
+* a method that encapsulates showing users the (non-wizard) confirmation prompt
+
+### Confirmation Prompt
+
+The confirmation prompt is a QuickPick instead of a message box.
+* The message box can be "dismissed" with the `Esc` key, which hides it in a "message box queue" rather than cancelling the message. The user flow appears to be cancelled, but in reality it is suspended (until the queued message box is restored, and a button is pressed).
+* QuickPick provides flexibility to add buttons to the UI (for example: a "Help" Button), and is appropriately cancel-able.
+
+### API Client
+
+The `ResourceGroupsTaggingAPI` service client will be used. The client will be set up in a manner consistent with the Lambda and CloudFormation clients, allowing the clients to be stubbed out in unit tests. See the Toolkit's Lambda Client [Interface](/src/shared/clients/lambdaClient.ts) and [Implementation](/src/shared/clients/defaultLambdaClient.ts) as an example. A new client factory will be added to [ToolkitClientBuilder](/src/shared/clients/toolkitClientBuilder.ts) and [DefaultToolkitClientBuilder](/src/shared/clients/defaultToolkitClientBuilder.ts).
+
+### Unit Tests
+
+* Deleting a Lambda Function
+  * it does not prompt the user with the pipeline-related confirmation when attempting to delete a Lambda Function that is not associated with a pipeline
+  * it prompts the user with the pipeline-related confirmation when attempting to delete a Lambda Function that is associated with a pipeline
+* Deleting a CloudFormation Stack
+  * it does not prompt the user with the pipeline-related confirmation when attempting to delete a CloudFormation Stack that is not associated with a pipeline
+  * it prompts the user with the pipeline-related confirmation when attempting to delete a CloudFormation Stack that is associated with a pipeline
+* Deploying a SAM Application
+  * it does not prompt the user with the pipeline-related confirmation when attempting to deploy a SAM Application to a CloudFormation Stack that is not associated with a pipeline
+  * it prompts the user with the pipeline-related confirmation when attempting to deploy a SAM Application to a CloudFormation Stack that is associated with a pipeline
```

Comment:
* it persistently prompts the user if they go back mid-workflow and reselect the same Cloudformation Stack.